great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1318,3 +1322,38 @@ def get_batch_data_and_markers(
             )
 
         return batch_data, batch_markers
+
+    def __del__(self):
+        """Close any open connections on shutdown."""
+        if self._connection:
+            self._connection.close()
+
+    # TODO: What is return type, docstring
+    def execute_query(self, query: Selectable):
+        """Execute a query using the underlying database engine.
+
+        Some databases sqlite/mssql temp tables only persist within a connection,
+        so we need to keep the connection alive.
+
+        Args:
+            query:
+
+        Returns:
+
+        """
+
+        # TODO: Add docstring
+
+        if self.dialect_name in _dialects_requiring_persisted_connection():
+            # sqlite/mssql temp tables only persist within a connection,
```

Comment:
nit: you can remove `sqlite/mssql` and refer to the function that returns these so this comment doesn't go out of date.

Reply:
Good call, changed to be more generic.